scss/_progressive-font.scss

```diff
@@ -0,0 +1,84 @@
+$_o-typography-progressive-font-fallbacks: (
+	sans: (
+		font: $o-typography-sans,
+		fallback: sans-serif,
+		fallback-scale: 0.87
+	),
+	sansBold: (
+		font: $o-typography-sans,
+		fallback: sans-serif,
+		fallback-scale: 0.83
+	),
+	sansData: (
+		font: $o-typography-sans,
+		fallback: sans-serif,
+		fallback-scale: 0.87
+	),
+	sansDataBold: (
+		font: $o-typography-sans,
+		fallback: sans-serif,
+		fallback-scale: 0.83
+	),
+	sansDataItalic: (
+		font: $o-typography-sans,
+		fallback: sans-serif,
+		fallback-scale: 0.87
+	),
+	serif: (
+		font: $o-typography-serif,
+		fallback: serif,
+		fallback-scale: 0.84
+	),
+	serifBold: (
+		font: $o-typography-serif,
+		fallback: serif,
+		fallback-scale: 0.84
+	),
+	serifItalic: (
+		font: $o-typography-serif,
+		fallback: serif,
+		fallback-scale: 0.84
+	),
+	serifDisplay: (
+		font: $o-typography-serif-display,
+		fallback: serif,
+		fallback-scale: 0.9
+	),
+	serifDisplayBold: (
+		font: $o-typography-serif-display,
+		fallback: serif,
+		fallback-scale: 0.9
+	),
+	serifDisplayItalic: (
+		font: $o-typography-serif-display,
+		fallback: serif,
+		fallback-scale: 0.9
+	)
+);
+
+@mixin oTypographyProgressiveFont($font, $level) {
+	$font-config: map-get($_o-typography-progressive-font-fallbacks, $font);
+	$is-progressive: index($o-typography-progressive-fonts, $font);
+
+	@if $is-progressive {
+		font-family: map-get($font-config, 'fallback');
+		@include font-size(get-font-size($font, $level, $is-fallback-font: true), get-line-height($font, $level));
+
+		.loaded-#{$font} & {
```

Comment:
`.loaded` should really be `.o-fonts--loaded` or configurable via `$o-fonts-loaded-prefix` sass variable